### ISSUE TITLE
Using '-o nospace' prevents adding spaces after quoted completion

### ISSUE THREAD

**Issue by evanunderscore** [kislyuk/argcomplete#151]
Since `quote_completions` doesn't add the trailing space when the input is quoted, nothing adds the space after the closing quote:

```
#!/usr/bin/env python
import argparse
import argcomplete
p = argparse.ArgumentParser()
p.add_argument('foo', choices=['foo', 'bar'])
argcomplete.autocomplete
```

```
$ test f<tab>
$ test foo <cursor>

$ test "f<tab>
$ test "foo"<cursor>

$ test 'f<tab>
$ test 'foo'<cursor>
```

This could be fixed by letting bash handle completions as described in #50.

**Comment by kislyuk** [comment]
If there are no side effects like what we found in #50, maybe we could add a space after a quoted completion if it's unique. With that said, having to press space to continue putting together the command line doesn't seem like a huge issue to me in this case.

**Comment by evanunderscore** [comment]
The space would have to be added after the closing quote, which is being inserted by bash. I experimented with getting argcomplete to add the closing quote but bash just adds another one after it so you end up with `"foo" "`.

Another option is to offload the nospace decision to the bash script in a similar fashion to http://stackoverflow.com/questions/8503126/custom-directory-completion-appends-whitespace/12185493#12185493.

```
if [[ "$COMPREPLY" =~ [/=:]$ ]]; then
    compopt -o nospace
fi
```

This would mean we can always return completions without spaces, which may also solve #50.

**Comment by evanunderscore** [comment]
This was fixed in #159
